### PR TITLE
Add method to check that collections has items and all items are not null/default

### PR DESCRIPTION
**Pull request by vsevolod-mozgovoi-am**
New methods: `HasAllNotNullItems` and `HasAllNotDefaultItems` which check that collections are not null, have items and all this items are not null/default

### CODE REVIEW

File: src/Directory.Packages.props [hunk 1]
```diff
@@ -8,13 +8,7 @@
...
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-    </PackageVersion>
-    <PackageVersion Include="SonarAnalyzer.CSharp" Version="9.26.0.92422">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-    </PackageVersion>
+    <PackageVersion Include="xunit.runner.visualstudio" Version="2.8.1" />
+    <PackageVersion Include="SonarAnalyzer.CSharp" Version="9.26.0.92422" />
   </ItemGroup>
 </Project>
```
Comment (IlyaBezus) [inline src/Directory.Packages.props]: could you fix that part too pls? :)

File: src/projects/EnsureThat/Enforcers/CollectionArg.cs [hunk 2]
```diff
@@ -406,5 +406,312 @@ public T[] HasAny<T>([ValidatedNotNull, NotNull]T[] value, Func<T, bool> predica
...
             return value;
         }
+
+        [return: NotNull]
+        [ContractAnnotation("value:null => halt")]
+        public ICollection<T> HasAllNotNullItems<T>([ValidatedNotNull, NotNull]ICollection<T> value, [InvokerParameterName, CallerArgumentExpression(nameof(value))] string paramName = null, OptsFn optsFn = null) where T : class
+        {
+            Ensure.Any.IsNotNull(value, paramName);
+
+            Ensure.Collection.HasItems(value, paramName);
```
Comment (IlyaBezus) [inline src/projects/EnsureThat/Enforcers/CollectionArg.cs]: why new line though?

Comment (vsevolod-mozgovoi-am) [reply src/projects/EnsureThat/Enforcers/CollectionArg.cs]: This method checks that:
1. Collection is not null
2. Collection has items
3. All items in collection are not null/default

**Review by IlyaBezus — APPROVED** [review]

File: src/projects/EnsureThat/Enforcers/CollectionArg.cs (hunk 2 above)
Comment (alexander-dyatlov-am) [reply src/projects/EnsureThat/Enforcers/CollectionArg.cs]: Is the ```IsNotNull``` check necessary? Since ```HasItems``` contains this check

**Review by alexander-dyatlov-am — APPROVED** [review]

**Pull request merged**